samcli/lib/package/regexpr.py

```diff
@@ -0,0 +1,4 @@
+"""
+Regular Expressions for Resources.
+"""
+ECR_URL = r"(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?\/.*"
```

Comment:
Just curious, is this documented anywhere? Or is it possible this will break some day?

Reply:
what we could document here is the docs section based on which the regex is derived.

https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_Repository.html does showcase the pattern for the repository name but not the entire regex.

What we can do is add a comment to link this above and also change the regex to allow for checking on the repository name as well.